Meta/run.py

```diff
@@ -0,0 +1,978 @@
+        elif config.machine_type == MachineType.QEMUExtLinux:
+            config.disk_image = "extlinux_disk_image"
+
+    if config.qemu_kind == QEMUKind.NativeWindows:
+        config.disk_image = run(
+            ["wslpath", "-w", config.disk_image], capture_output=True, encoding="utf-8"
+        ).stdout.strip()
+
+
+def setup_cpu(config: Configuration):
+    if config.qemu_kind == QEMUKind.NativeWindows:
+        config.qemu_cpu = "max,vmx=off"
+    else:
+        provided_cpu = environ.get("SERENITY_QEMU_CPU")
+        if provided_cpu is not None:
+            config.qemu_cpu = provided_cpu
+
+
+def setup_cpu_count(config: Configuration):
+    if config.architecture == Arch.Aarch64:
+        return
+
+    provided_cpu_count = environ.get("SERENITY_CPUS")
+    if provided_cpu_count is not None:
+        try:
+            config.cpu_count = int(provided_cpu_count)
+        except ValueError:
+            raise RunError(f"Non-integer CPU count {provided_cpu_count}")
+
+    if config.cpu_count <= 8:
+        config.qemu_cpu += ",-x2apic"
+
+
+def setup_spice(config: Configuration):
+    if environ.get("SERENITY_SPICE") == "1":
+        chardev_info = run(
+            [cast(str, config.qemu_binary), "-chardev", "help"],
+            capture_output=True,
+            encoding="utf-8",
+        ).stdout.lower()
+        if "qemu-vdagent" in chardev_info:
+            config.spice_arguments = [
+                "-chardev",
+                "qemu-vdagent,clipboard=on,mouse=off,id=vdagent,name=vdagent",
+            ]
+        elif "spicevmc" in chardev_info:
+            config.spice_arguments = ["-chardev", "spicevmc,id=vdagent,name=vdagent"]
+        else:
+            raise RunError("No compatible SPICE character device was found")
+
+        if "spice" in chardev_info:
+            config.spice_arguments.extend(
+                ["-spice", "port=5930,agent-mouse=off,disable-ticketing=on"]
+            )
+        if "spice" in chardev_info or "vdagent" in chardev_info:
+            config.spice_arguments.extend(
+                ["-device", "virtserialport,chardev=vdagent,nr=1"]
+            )
+
+
+def setup_audio_backend(config: Configuration):
+    if config.qemu_kind == QEMUKind.MacOS:
+        config.audio_backend = "coreaudio"
+    elif config.qemu_kind == QEMUKind.NativeWindows:
+        config.audio_backend = "dsound,timer-period=2000"
+    else:
+        # FIXME: Use "-audiodev help" once that contains information, "-audio-help" is marked as deprecated.
+        qemu_audio_help = run(
+            [cast(str, config.qemu_binary), "-audio-help"],
+            capture_output=True,
+            encoding="utf-8",
+        ).stdout
+        if "-audiodev id=sdl" in qemu_audio_help:
+            config.audio_backend = "sdl"
+
+    config.audio_backend += ",id=snd0"
+
+
+def setup_audio_hardware(config: Configuration):
+    provided_audio_hardware = environ.get("SERENITY_AUDIO_HARDWARE") or "intelhda"
```

Comment:
get() has a second argument for a default 
```
>>> os.environ.get("FOO", "Default")
'Default'
```

Reply:
Yeah, although all of these calls should be replaced by calls to a user config input handler sooner than later.